### PR TITLE
Cloud migration UI: Add `migrate-to-cloud` route

### LINKED ISSUE #82313: E2C: Create new frontend route for e2c workflow
Feature toggle: `onPremToCloudMigrations`

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- adds a new `Migrate to Grafana Cloud` nav item under `Admin -> General`
  - gated behind the `onPremToCloudMigrations` feature toggle
  - also gated behind an access control check
- adds translations for the new nav item
- adds a new `MigrateToCloud` page that is currently very empty 😄 
- adds a new `/admin/migrate-to-cloud` route pointing to this new page
  - gated behind the `onPremToCloudMigrations` feature toggle

**Why do we need this feature?**

- help people migrate to cloud!

**Who is this feature for?**

- everyone on prem

**Which issue(s) does this PR fix?**:

Fixes #82313 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/admin/migrate-to-cloud/MigrateToCloud.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+import { Page } from 'app/core/components/Page/Page';
+
+export default function MigrateToCloud() {
+  return <Page navId="migrate-to-cloud">TODO</Page>;
```
Comment (joshhunt) [inline public/app/features/admin/migrate-to-cloud/MigrateToCloud.tsx:6]: :shipit:

File: pkg/services/navtree/navtreeimpl/admin.go [hunk 2]
```diff
@@ -135,6 +135,16 @@ func (s *ServiceImpl) getAdminNode(c *contextmodel.ReqContext) (*navtree.NavLink
 		configNodes = append(configNodes, storage)
 	}
 
+	if s.features.IsEnabled(ctx, featuremgmt.FlagOnPremToCloudMigrations) && hasAccess(ac.EvalPermission(ac.ActionSettingsRead, ac.ScopeSettingsAll)) {
```
Comment (ashharrison90) [inline pkg/services/navtree/navtreeimpl/admin.go:138]: is this the best way to check for server admin privileges? 🤔

File: public/app/routes/routes.tsx [hunk 3]
```diff
@@ -362,6 +362,12 @@ export function getAppRoutes(): RouteDescriptor[] {
         () => import(/* webpackChunkName: "ServerStats" */ 'app/features/admin/ServerStats')
       ),
     },
+    {
+      path: '/admin/migrate-to-cloud',
+      component: SafeDynamicImport(
+        () => import(/* webpackChunkName: "ServerStats" */ 'app/features/admin/migrate-to-cloud/MigrateToCloud')
+      ),
+    },
```
Comment (joshhunt) [inline public/app/routes/routes.tsx:370]: I think we should be in the habit of gating routes behind the feature flag as well.

File: pkg/services/navtree/navtreeimpl/admin.go (hunk 2 above)
Comment (IevaVasiljeva) [reply pkg/services/navtree/navtreeimpl/admin.go:138]: I think the best way would be to either check the permissions to all impacted components (eg, dashboards, data sources etc) or introduce a new permission for migration. But if you want to keep it simple and just check for server admin rights, use `hasRole('ServerAdmin')` (from [here](https://github.com/grafana/grafana/blob/main/public/app/core/services/context_srv.ts#L120-L126)) instead.

File: public/app/routes/routes.tsx [hunk 4]
```diff
@@ -362,6 +362,12 @@ export function getAppRoutes(): RouteDescriptor[] {
         () => import(/* webpackChunkName: "ServerStats" */ 'app/features/admin/ServerStats')
       ),
     },
+    config.featureToggles.onPremToCloudMigrations && {
+      path: '/admin/migrate-to-cloud',
+      component: SafeDynamicImport(
+        () => import(/* webpackChunkName: "MigrateToCloud" */ 'app/features/admin/migrate-to-cloud/MigrateToCloud')
```
Comment (IevaVasiljeva) [inline public/app/routes/routes.tsx:368]: I'd also add a permission check here. Something like
```
roles: () => ['ServerAdmin'],
```
or
```
roles: () => contextSrv.evaluatePermission([<some_permission>]),
```

There are a bunch of examples of that for other routes if you look around.

**Review by IevaVasiljeva — COMMENTED** [review]
Added some recommendations for how to do access checks.

**Review by joshhunt — APPROVED** [review]

**Pull request merged**